File: src/main/java/org/micoli/php/symfony/cliDumpParser/actions/ParseCliDumperToPhpAction.java
package org.micoli.php.symfony.cliDumpParser.actions;

import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import org.jetbrains.annotations.NotNull;
import org.micoli.php.symfony.cliDumpParser.JsonToPhpArrayConverter;
import org.micoli.php.symfony.cliDumpParser.PhpDumpHelper;
import org.micoli.php.ui.popup.ParsedContentDisplayPopup;

public class ParseCliDumperToPhpAction extends AbstractParseCliDumperAction {

    protected void parseCliDumperOutput(String text, Project project) {
        try {
            String parsed = JsonToPhpArrayConverter.convertJsonToPhp(PhpDumpHelper.parseCliDumperToJson(text), "   ");

            ParsedContentDisplayPopup.showPhpPopup(project, parsed);
        } catch (Exception ex) {
            Messages.showErrorDialog(project, "Error: \n" + ex.getMessage(), "Parse CliDumper");
        }
    }

    @Override
    protected void setPresentationText(@NotNull Presentation presentation, boolean isVisible) {
        presentation.setText(isVisible ? "Parse Symfony CliDumper to PHP" : "Parse Symfony CliDumper - No Selection");
    }
}

File: src/main/kotlin/org/micoli/php/symfony/cliDumpParser/actions/ParseCliDumperToPhpAction.kt
package org.micoli.php.symfony.cliDumpParser.actions

import com.intellij.openapi.actionSystem.Presentation
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.Messages
import org.micoli.php.symfony.cliDumpParser.JsonToPhpArrayConverter
import org.micoli.php.symfony.cliDumpParser.PhpDumpHelper
import org.micoli.php.ui.popup.ParsedContentDisplayPopup.Companion.showPhpPopup

class ParseCliDumperToPhpAction : AbstractParseCliDumperAction() {
    override fun parseCliDumperOutput(text: String, project: Project) {
        try {
            val jsonString = PhpDumpHelper.parseCliDumperToJson(text) ?: return
            val parsed = JsonToPhpArrayConverter.convertJsonToPhp(jsonString, "   ")

            showPhpPopup(project, parsed)
        } catch (ex: Exception) {
            Messages.showErrorDialog(project, "Error: \n" + ex.message, "Parse CliDumper")
        }
    }

    override fun setPresentationText(presentation: Presentation, isVisible: Boolean) {
        presentation.setText(
            if (isVisible) "Parse Symfony CliDumper to PHP"
            else "Parse Symfony CliDumper - No Selection")
    }
}
